crates/ruma-common/src/events/room/redaction.rs

```diff
@@ -355,46 +350,24 @@ impl From<RoomRedactionEvent> for SyncRoomRedactionEvent {
 }
 
 impl OriginalRoomRedactionEvent {
-    /// Returns the ID of the event that this event redacts, according to the given room version.
+    /// Returns the ID of the event that this event redacts, according to the proper `redacts` field
+    /// for the given room version.
     ///
-    /// Returns `None` if the only available `redacts` field is not the proper one for the given
-    /// room version.
-    pub fn redacts(&self, room_version: &RoomVersionId) -> Option<&EventId> {
-        match room_version {
-            RoomVersionId::V1
-            | RoomVersionId::V2
-            | RoomVersionId::V3
-            | RoomVersionId::V4
-            | RoomVersionId::V5
-            | RoomVersionId::V6
-            | RoomVersionId::V7
-            | RoomVersionId::V8
-            | RoomVersionId::V9
-            | RoomVersionId::V10 => self.redacts.as_deref(),
-            _ => self.content.redacts.as_deref(),
-        }
+    /// If the `redacts` field is not the proper one for the given room version, this falls back to
+    /// the one that is available.
+    pub fn redacts(&self, room_version: &RoomVersionId) -> &EventId {
+        redacts(room_version, self.redacts.as_deref(), self.content.redacts.as_deref())
     }
 }
 
 impl OriginalSyncRoomRedactionEvent {
-    /// Returns the ID of the event that this event redacts, according to the given room version.
+    /// Returns the ID of the event that this event redacts, according to the proper `redacts` field
+    /// for the given room version.
     ///
-    /// Returns `None` if the only available `redacts` field is not the proper one for the given
-    /// room version.
-    pub fn redacts(&self, room_version: &RoomVersionId) -> Option<&EventId> {
-        match room_version {
-            RoomVersionId::V1
-            | RoomVersionId::V2
-            | RoomVersionId::V3
-            | RoomVersionId::V4
-            | RoomVersionId::V5
-            | RoomVersionId::V6
-            | RoomVersionId::V7
-            | RoomVersionId::V8
-            | RoomVersionId::V9
-            | RoomVersionId::V10 => self.redacts.as_deref(),
-            _ => self.content.redacts.as_deref(),
-        }
+    /// If the `redacts` field is not the proper one for the given room version, this falls back to
+    /// the one that is available.
+    pub fn redacts(&self, room_version: &RoomVersionId) -> &EventId {
```

Comment:
I'm a bit worried about panicking instead of returning `None` if we don't find `redacts`. I think after this PR, somebody might be able to deserialize a redacted redaction event as `OriginalRoomRedactionEvent`, and then when calling this method it would panic right? (it would also be possible for somebody to assign `None` to both fields in Rust code, but that seems less likely)

Reply:
Ah no I see what you mean

Reply:
We should implement a custom deserialize for `OriginalRoomRedactionEvent` instead of handling it in `RoomRedactionEvent`

Reply:
Is that better?

Reply:
Haven't looked at the impl yet, but I think your idea sounds good. Can you also document that these functions can panic (in weird circumstances where ppl have set both fields to `None` from Rust code)?

Reply:
Done